src/macaron/slsa_analyzer/provenance/loader.py

```diff
@@ -52,3 +54,46 @@ def load(cls, path: str) -> Any:
             raise SLSAProvenanceError(
                 f"Cannot decode the message content of the SLSA attestation - {error.reason}"
             ) from error
+
+
+def load_provenance(filepath: str) -> Any:
```

Comment:
This method seems to duplicate `ProvPayloadLoader.load` method?

Reply:
I didn't want to use the original `ProvPayloadLoader.load` methods for a couple of reasons. 
* The error messages of this method mention SLSA provenances but I was loading witness provenances rather than SLSA provenances.
* I also wanted the return type to be `JsonType` instead of `Any`, but I don't want to change the return type of the `ProvPayloadLoader.load` method since it is being used in the provenance level 3 check. Fixing the return type will result in type errors that I don't want to fix in this PR.
If you agree then I can create an issue to refactor these two methods of loading provenances into one.

Reply:
I think the `SLSAProvenanceError` is not quite accurate in the original implementation. Based on the [in-toto spec](https://github.com/in-toto/attestation/blob/main/spec/v1/envelope.md), it would make more sense to use `IntotoAttestationError` for in-toto attestations instead and remove `load_provenance`.

Reply:
I have refactored loading and validating provenances in e519b9ab432f4a86b1824731d3d399682b816170.